File: archlucid-ui/src/lib/review-quality/authority-commit-skipped-must-blocked-reason.ts
/** Canonical skipped-MUST seal blocked reason (server source of truth mirror — WA-12). */
export function formatAuthorityCommitSkippedMustBlockedReason(skippedMustCount: number): string {
  if (skippedMustCount <= 0) {
    throw new RangeError("skippedMustCount must be positive");
  }

  const noun = skippedMustCount === 1 ? "question is" : "questions are";

  return `${skippedMustCount} required ${noun} unanswered.`;
}

File: ArchLucid.Application/Runs/Orchestration/Commit/AuthorityCommitSkippedMustBlockedReason.cs
namespace ArchLucid.Application.Runs.Orchestration.Commit;

/// <summary>Canonical skipped-MUST blocked reason shared by API, CLI, and UI scorecard (WA-12).</summary>
public static class AuthorityCommitSkippedMustBlockedReason
{
    public static string Format(int skippedMustCount)
    {
        if (skippedMustCount <= 0)
        {
            throw new ArgumentOutOfRangeException(nameof(skippedMustCount));
        }

        string noun = skippedMustCount == 1 ? "question is" : "questions are";

        return $"{skippedMustCount} required {noun} unanswered.";
    }
}
